src/routes/blog/post/announcing-more-and-updated-runtimes/+page.markdoc

```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title: Introducing more and updated runtimes to the Appwrite Cloud ecosystem
+description: More runtimes, more flexibility. We're happy to add more and updated runtimes to allow you to build with your preferred tech stack.
+date: 2024-03-01
+cover: /images/blog/runtimes.png
+timeToRead: 6
+author: vincent-ge
+category: product
+---
+
+Previously, we completely reimagined [Functions](https://dev.to/appwrite/serverless-your-way-unleashing-appwrite-functions-true-potential-2l4f) to be more flexible and innovative yet familiar to developers. Now, Appwrite expands and updates the runtime ecosystem available on Appwrite Cloud with Bun 1.0.29, Node 21, Ruby 3.3, Deno 1.40, PHP 8.3, Python 3.12, Kotlin 1.9, Java 18, Swift 5.9 and Dart 3.3 support.
+
+We’re excited to have updated versions so you can build with your preferred tech stack and Appwrite. Let’s dive into some of these runtimes and why they are a great addition to the Cloud runtime ecosystem.
+
+# Bun, blazing fast and built for web
+
+[Bun](https://bun.sh/) is an amazingly fast runtime, dependency manager, test runner, and web server for JavaScript with unique features that make it a perfect fit for Appwrite Functions. Bun is fast, like really fast. The runtime performance, dependency manager, and test runners are noticeably faster than its many Node.js counterparts. This is a topic well discussed in [benchmarks](https://medium.com/deno-the-complete-reference/is-bun-really-much-faster-than-node-js-e5b15942a8e8). 
+
+Bun also boasts a robust standard library. It enables TypeScript and JSX out of the box without a build step, implements standard Web APIs like `fetch()`, and comes with a built-in test runner. All this reduces build times and dependencies needed. Our Bun 1.0.29 runtime brings improved compatibility with the NPM ecosystem and countless bug fixes.
+
+# Dart 3.3, sound null safety
+
+Dart is one of Appwrite’s [fastest runtimes](https://dev.to/appwrite/take-your-serverless-functions-to-new-speeds-with-appwrite-013-5868) because it is compiled with impressive cold-start performance and millisecond response times when warm. For Flutter developers, you can write your app and Appwrite Functions code in Dart, track them in the same repository, and deploy your app and function together when you PR to your production branch.
+
+For Dart developers, Dart 3 introduces many exciting features like records for structured data, patterns, more verbose OOP concepts, and more. For Appwrite Functions developers, however, sound null safety is a stand-out.
+
+Sound null safety means unless you specify a variable can be null, it will **never be null.** At compile time, not accessing nullable values safely or assigning potentially null values to non-nullable variables will throw errors.
```

Comment:
```suggestion
Sound null safety means unless you specify a variable can be null, it will **never be null**. At compile time, not accessing nullable values safely or assigning potentially null values to non-nullable variables will throw errors.
```